Mathlib/Tactic/ToAdditive.lean

```diff
@@ -288,6 +288,11 @@ partial def transformDeclAux
       Failed to add declaration\n{trgDecl.name}:\n{msg}"
     | _ => panic! "unreachable"
   addAndCompile trgDecl.toDeclaration!
+  -- now add declaration ranges so jump-to-definition works
+  addDeclarationRanges tgt {
+    range := ← getDeclarationRange ref
+    selectionRange := ← getDeclarationRange (getDeclarationSelectionRef ref)
```

Comment:
```suggestion
    range := ← getDeclarationRange (← getRef)
    selectionRange := ← getDeclarationRange ref
```
The `getRef` should be set to the enclosing command (I think this is the default behavior).

Also you need a `Term.addTermInfo'` on the same span.

Reply:
Unfortunately `Term.addTermInfo'` requires us to be in `TermElabM`, and throughout this code we are only in `AttrM := CoreM`.

Reply:
yes, you need a `liftTermElabM` as in `alias`

Reply:
I don't think `liftTermElabM` helps: that takes something in `TermElabM` and puts it in `CommandElabM`, but we need to be in `CoreM` here.

Reply:
You can use `TermElabM.run'` and `MetaM.run'` to get from CoreM to TermElabM

Reply:
Okay, that's working now, and we get a hover. Curiously the hover only shows when we are over the to_additive doc-string, or the alternate name, but not over the characters `to_additive` themselves.